mlflow/store/artifact/runs_artifact_repo.py

```diff
@@ -102,24 +107,77 @@ def log_artifacts(self, local_dir, artifact_path=None):
     def _is_directory(self, artifact_path):
         return self.repo._is_directory(artifact_path)
 
-    def list_artifacts(self, path):
+    def list_artifacts(self, path: Optional[str] = None) -> list[FileInfo]:
         """
         Return all the artifacts for this run_id directly under path. If path is a file, returns
-        an empty list. Will error if path is neither a file nor directory.
+        an empty list. Will error if path is neither a file nor directory. When the run has an
+        associated model, the artifacts of the model are also listed.
 
         Args:
             path: Relative source path that contain desired artifacts
 
         Returns:
             List of artifacts as FileInfo listed directly under path.
         """
+        return self._list_run_artifacts(path) + self._list_model_artifacts(path)
```

Comment:
@harupy Just confirming, everything works fine if there's a run artifact called `foo` and a model artifact called `foo`?